webcompat/webhooks/helpers.py

```diff
@@ -106,8 +108,12 @@ def extract_priority_label(body):
     hostname = domain_name(extract_url(body))
     if hostname:
         priorities = ['critical', 'important', 'normal']
-        # Find host_name in DB
-        for site in site_db.query(Site).filter_by(url=hostname):
+        # Find host_name in regional DB first,
+        # if it doesn't exist, look in global
+        for site in regional_site_db.query(SiteRegional).filter_by(url=hostname):       # noqa
+            return f'priority-{priorities[site.priority - 1]}'
```

Comment:
I don't remember why the code is using a for loop. Maybe an opportunity to fix it. 

Do we expect `regional_site_db.query(SiteRegional).filter_by(url=hostname)` to return more than one domain? and it would mean that even if there was more than one domain, only the first hit would be taken into consideration, because of the return.

Basically the code is currently the equivalent of 

```python
sites = ['A', 'B', 'C', 'A2', 'C']
def extract(sites):
    for site in sites:
        if site.startswith('A'):
            return site
```

which would return 'A' always.

Should we just have something like.

```suggestion
        site = regional_site_db.query(SiteRegional).filter_by(url=hostname).first()
        if not site:
            site = global_site_db.query(SiteGlobal).filter_by(url=hostname).first()
        if site:
            return f'priority-{priorities[site.priority - 1]}'
        #… and so on for the subdomains.   
```

`.first()` returns None if the row doesn't exist.

or something else. 


Maybe there is a function to extract here. because the code is repeating the code in a loop. 

Or maybe I totally misunderstood the code 🍭 Long time I have touched this part.

Reply:
The domain names are unique, so using `.first()` will work indeed. Thanks for the suggestion, I've changed the function